tests/_pure_is_inside_container.test.fish

```diff
@@ -53,12 +53,25 @@ end
     echo $called
 ) = once
 
-cleanup_spy
-@test "_pure_is_inside_container: true for Github Action" (
-    set --universal pure_enable_container_detection true
+if test (uname -s) != Darwin
+    cleanup_spy
+    @test "_pure_is_inside_container: true for Github Action on Ubuntu" (
+        set --universal pure_enable_container_detection true
 
-    _pure_is_inside_container
-) $status -eq $SUCCESS
+        _pure_is_inside_container
+    ) $status -eq $SUCCESS
+    end
```

Comment:
I introduce an extraneous `end` that causes the [tests to fail](https://github.com/pure-fish/pure/actions/runs/7635299204/job/20807724766?pr=340#step:5:104)
```suggestion
```